preliminary-work/nematic_order.py

```diff
@@ -0,0 +1,40 @@
+import flowermd
+import hoomd
+import gsd
+import numpy as np
+import freud
+
+
+def nematic_order(gsd_file, start=0, stop=None, stride=1):
+      """
+       Calculate nematic order for a coarse-grained system by calculating bond vectors for each bond between two beads.
+       
+       Parameters
+        ----------
+        gsdfile : str
+            Filename for gsd file
+        start : int, default 0
+            Where to start reading the gsd trajectory the system was created
+            with.
+        stop : int, default None
+            Where to stop reading the gsd trajectory the system was created
+            with. If None, will stop at the last frame.
+        stride : int, default 1
+            The step size to use when iterating through start:stop
+        """
+    nm_orders = []
+    directors = [] #adding temporary directors list to export directors generated from nematic function
+    with gsd.hoomd.open(gsd_file) as traj: #opening gsd file as traj
+        for snap in traj[start:stop:stride]: #looping through each frame
+            bond_vectors = []
+            nematic = freud.order.Nematic() 
```

Comment:
Pass in the director parameter here.

Reply:
I tried this and a few other methods and ultimately got this error:

      property of 'Nematic' object has no setter

I tried looking through the code and I'm not sure how we would set this.

Reply:
Can you show what you tried and paste the entire error?

We should pass in the director parameter with something like:

```
 nematic = freud.order.Nematic(np.asarray(director))
```

Reply:
Love seeing this activity!!

Reply:
So I tried this:

```
def nematic_order(gsd_file, director, start=0, stop=None, stride=1):
    nm_orders = []
    directors = [] #adding temporary directors list to export directors generated from nematic function
    with gsd.hoomd.open(gsd_file) as traj: #opening gsd file as traj
        for snap in traj[start:stop:stride]: #looping through each frame
            bond_vectors = []
            nematic = freud.order.Nematic()
            **nematic.director = director**
            for bond in snap.bonds.group: #looping through each bond in frame
                pos2 = snap.particles.position[bond[1]]
                pos1 = snap.particles.position[bond[0]]
                vec = pos2 - pos1
                bond_vectors.append(vec)
            nematic.compute(np.array(bond_vectors))#computing nematic order for each frame
            nm_orders.append(nematic.order)
            print(nematic.director)
            directors.append(nematic.director)
    
    return(nm_orders)
```
Passing in:
```
director = np.array([0,1,0])
lattice_orders = nematic_order("lattice-cg.gsd",director)
```
and got this error:  property of 'Nematic' object has no setter

I first tried this:

```
      nm_orders = []
      directors = [] #adding temporary directors list to export directors generated from nematic function
      with gsd.hoomd.open(gsd_file) as traj: #opening gsd file as traj
          for snap in traj[start:stop:stride]: #looping through each frame
              bond_vectors = []
              **nematic = freud.order.Nematic(np.asarray(director))**
...
      
```
This isn't giving me an error but I don't think it's working. There is no change when I change the director being passed in and the list of directors is random and not set.

I also tried passing it in as:

`nematic.compute(director, np.array(bond_vectors))`
or
`nm_orders.append(nematic.order(director))`
           
nematic.compute only takes one argument. And adding it as an argument to nematic.order gives this error: 'float' object is not callable.

Reply:
Ok, the 2nd way  you mention here would be the correct way to pass it in, but now that I'm looking at it more, I think I was wrong that we can set a director that is used as the reference vector. I tried a bunch of different options, and it doesn't seem to make any difference. So, lets not include it as a parameter.